### PR TITLE
VmdkReadHandle: gracefully shutdown and close the socket

### PR DESCRIPTION
**Pull request by leust**
There is an additional issue that causes orphaned VMDKs, that the `close()` doesn't actually close the connection immediately, but we must first call `.shutdown()` first.
>close() releases the resource associated with a connection
but does not necessarily close the connection immediately.
If you want to close the connection in a timely fashion,
call shutdown() before close().

This works in conjunction with the [.sleep(5) workaround](https://github.com/sapcc/oslo.vmware/pull/31), which is needed to wait for VMware to release the lock on the VMDK once the http connection is (properly) closed.

### CODE REVIEW

**Comment by leust** [comment]
This was tested in QA environment and works well, the VMDK being deleted successfully when a backup is interrupted.

File: oslo_vmware/rw_handles.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 """
 
 import logging
+import socket
+
```
Comment (joker-at-work) [inline oslo_vmware/rw_handles.py:26]: I don't think this should add a newline and thus a new block of imports.

File: oslo_vmware/rw_handles.py [hunk 2]
```diff
@@ -721,13 +723,26 @@ def read(self, chunk_size):
     def tell(self):
         return self._bytes_read
 
+    def _shutdown_connection(self):
+        """Gracefully shuts down the socket of the HTTP(S) connection"""
+        # HTTPSConnection wraps the TCP socket into an SSL transport,
+        # which needs to be shutdown first.
+        if 'WrappedSocket' in type(self._conn.sock).__name__:
+            _socket = self._conn.sock.socket
+            self._conn.sock.shutdown()
```
Comment (joker-at-work) [inline oslo_vmware/rw_handles.py:732]: Trying to find out how the interface of `shutdown()` looks like here, I cannot find a `WrappedSocket` object (I've greped through my local Python installation for that string, Python 3.8). Afaics, it should be an `SSLSocket` object [`wrap_socket()` docs](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.wrap_socket), which seems to have the same interface for `shutdown()` and as I understand it should call the socket's `shutdown()` [docs](https://docs.python.org/3/library/ssl.html#ssl.SSLSocket)

Can you help me understand this, please?

**Review by fwiesel — APPROVED** [review]

File: oslo_vmware/rw_handles.py (hunk 1 above)
Comment (leust) [reply oslo_vmware/rw_handles.py]: Fixed. It was added like this by the IDE.

File: oslo_vmware/rw_handles.py (hunk 2 above)
Comment (joker-at-work) [reply oslo_vmware/rw_handles.py:731]: thank you.

**Review by joker-at-work — APPROVED** [review]

**Review by hemna — APPROVED** [review]

**Pull request merged**